Xamarin.Forms.Platform.Android/CollectionView/ObservableGroupedSource.cs

```diff
@@ -232,26 +231,29 @@ void ClearGroupTracking()
 
 		void CollectionChanged(object sender, NotifyCollectionChangedEventArgs args)
 		{
-			switch (args.Action)
+			Device.BeginInvokeOnMainThread(() =>
```

Comment:
This should be checking IsInvokeRequired first; if we're already on the main thread, there's no need to queue this up on the thread.

Reply:
Good point. Changed!.